### PR TITLE
feat: add new non-native Modal component

### PR DESCRIPTION
**Pull request by acezard**
This is needed on screens where we do not have agency in what is displayed on top of the screen. While it may be tempting to use it everywhere for consistency, I believe it's better to rely on the react-native Modal component when possible. This should be used as a last resort where FullWindowOverlay for instance disallow us to display anything above it (on the native UI side).

### CODE REVIEW

File: src/ui/CozyDialogs/ConfirmDialog.tsx [hunk 1]
```diff
@@ -61,14 +98,56 @@ export const ConfirmDialog = ({
 
         <View style={styles.content}>
           {typeof content === 'string' ? (
-            <Typography>{content}</Typography>
```
Comment (acezard) [inline src/ui/CozyDialogs/ConfirmDialog.tsx:64]: the text is lighter than before here, is it intended? 
image

File: src/ui/CozyTheme/CozyTheme.tsx [hunk 2]
```diff
@@ -10,17 +10,22 @@ interface CozyThemeState {
...
 interface CozyThemeProps {
-  variant: CozyThemeVariant
+  variant?: CozyThemeVariant
   children: ReactNode
 }
 
+// Will default to `normal` variant if none is provided
 export const CozyTheme = ({
-  variant,
+  variant = 'normal',
```
Comment (Ldoppea) [inline src/ui/CozyTheme/CozyTheme.tsx:28]: I would prefer to have this not optional, as I find `<CozyTheme variant="normal">` easier to understand when reading the code compared to `<CozyTheme>` and the gain is significant enough to justify it.

File: src/ui/CozyDialogs/ConfirmDialog.tsx [hunk 3]
```diff
@@ -41,34 +71,72 @@ const renderActions = (
...
 }
 
+const ModalContent = ({
+  onClose,
+  title,
+  headerStyle,
+  content,
+  actions
+}: Partial<ConfirmDialogProps>): JSX.Element => (
+  <CozyTheme>
```
Comment (Ldoppea) [inline src/ui/CozyDialogs/ConfirmDialog.tsx:81]: If we put `<CozyTheme>` here this means that we don't expect to have `inverted` modals in the future. Maybe we should put it in the caller side? (LockScreen)

File: src/ui/Typography/index.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ type TypographyVariant =
   | 'underline'
   | 'link'
 
-type TypographyColor =
+export type TypographyColor =
```
Comment (Ldoppea) [inline src/ui/Typography/index.tsx:25]: You export this but it seems like you don't use it. Is that on purprose?

File: src/ui/CozyTheme/CozyTheme.tsx (hunk 2 above)
Comment (acezard) [reply src/ui/CozyTheme/CozyTheme.tsx:28]: ✅

File: src/ui/CozyDialogs/ConfirmDialog.tsx (hunk 3 above)
Comment (acezard) [reply src/ui/CozyDialogs/ConfirmDialog.tsx:81]: Agreed ✅

File: src/ui/Typography/index.tsx (hunk 4 above)
Comment (acezard) [reply src/ui/Typography/index.tsx:25]: ✅

**Review by Ldoppea — APPROVED** [review]

**Pull request merged**